src/ecs/resources/input.rs

```diff
@@ -1,24 +1,35 @@
 //! The input handler for the game engine
 
-use std::collections::hash_map::{Entry, HashMap};
+use std::collections::hash_map::{Entry, HashMap, Keys};
+use std::iter::Iterator;
 use event::{ElementState, WindowEvent, Event, VirtualKeyCode};
 
 #[derive(PartialEq, Eq)]
-enum KeyQueryState{
+enum KeyQueryState {
     NotQueried,
     Queried,
 }
 
+// Iterator that iterates through all currently pressed down keys
```

Comment:
This should be made into a doc comment (`///`) so it will appear in the documentation.